applications/DEMApplication/custom_elements/rigid_body_element.cpp

```diff
@@ -137,6 +137,20 @@ namespace Kratos {
                 central_node.FastGetSolutionStepValue(EXTERNAL_APPLIED_MOMENT)[2] = rigid_body_element_sub_model_part[EXTERNAL_APPLIED_MOMENT][2];
             }
 
+            if (central_node.Is(DEMFlags::FIXED_VEL_X) && central_node.Is(DEMFlags::FIXED_VEL_Y) && central_node.Is(DEMFlags::FIXED_VEL_Z) && central_node.Is(DEMFlags::FIXED_ANG_VEL_X) && central_node.Is(DEMFlags::FIXED_ANG_VEL_Y) && central_node.Is(DEMFlags::FIXED_ANG_VEL_Z)) {
+                if (rigid_body_element_sub_model_part.Has(COMPUTE_FORCES_ON_THIS_RIGID_ELEMENT)) {
+                    if (rigid_body_element_sub_model_part[COMPUTE_FORCES_ON_THIS_RIGID_ELEMENT]== 1) {
+                        central_node.Set(DEMFlags::COMPUTE_REACTIONS, true);
```

Comment:
Doing this in Initialize is kind of dangerous. What happens if the user releases the boundary conditions later on? We should probably check this at every time step.

Reply:
Ok, now it is done in the strategy, at each time step. I also consider that if the submodelpart is FORCE_INTEGRATION_GROUP, forces should be calculated as we need them.

Reply:
True, we should add this, too.